### PR TITLE
OCPBUGS-33594: trigger overall sync-state events

### PR DESCRIPTION
**Pull request by jzding**
trigger overall sync-state event when either ptp-state-change or os-clock-sync-state-change events are triggered.

### CODE REVIEW

File: plugins/ptp_operator/metrics/manager.go [hunk 1]
```diff
@@ -261,10 +261,19 @@ func (p *PTPEventManager) PublishEvent(state ptp.SyncState, ptpOffset int64, sou
...
 		return
 	}
+
 	// /cluster/xyz/ptp/CLOCK_REALTIME this is not address the event is published to
 	data := p.GetPTPEventsData(state, ptpOffset, source, eventType)
 	resourceAddress := fmt.Sprintf(p.resourcePrefix, p.nodeName, string(p.publisherTypes[eventType].Resource))
 	p.publish(*data, resourceAddress, eventType)
+	// publish the event again as overall sync state
+	// SyncStateChange is the overall sync state including PtpStateChange and OsClockSyncStateChange
+	if eventType == ptp.PtpStateChange || eventType == ptp.OsClockSyncStateChange {
```
Comment (aneeshkp) [inline plugins/ptp_operator/metrics/manager.go:271]: you will end up sending twice or more time . when there is ptpState Change event in OC  or BC, there will be following OS clock event , not the other way .
So 1) send overall state when there is only OSClock for OC/BC and 2) send overall State for BC without Phc2sys in its Profile.
3) Also for HA ,send Over All state when OSCLOCK sends event
FO T_-GM it is handled differently  OSClock sate don't impact T-GM state but tsphc and DPLL impacts T-GM state

File: plugins/ptp_operator/ptp_operator_plugin.go [hunk 2]
```diff
@@ -245,14 +245,16 @@ func getCurrentStatOverrideFn() func(e v2.Event, d *channel.DataChan) error {
...
 						clockClass := fmt.Sprintf("%s/%s", string(ptpInterface), ptpMetrics.ClockClass)
 						data = processDataFn(data, eventManager.GetPTPEventsData(s.SyncState(), s.ClockClass(), clockClass, eventType))
-					// overallSyncState is only for master
 					case ptp.SyncStateChange:
 						overallSyncState = getOverallState(overallSyncState, s.SyncState())
 					}
 				case ptpMetrics.ClockRealTime:
 					switch eventType {
 					case ptp.OsClockSyncStateChange:
 						data = processDataFn(data, eventManager.GetPTPEventsData(s.SyncState(), s.LastOffset(), string(ptpInterface), eventType))
```
Comment (aneeshkp) [inline plugins/ptp_operator/ptp_operator_plugin.go:254]: check line #472 and #488 when ptpconfig delete is detected , an event is sent out

Comment (jzding) [reply plugins/ptp_operator/ptp_operator_plugin.go:254]: that is covered in line #272-279 in plugins/ptp_operator/metrics/manager.go

File: plugins/ptp_operator/metrics/manager.go (hunk 1 above)
Comment (jzding) [reply plugins/ptp_operator/metrics/manager.go:272]: updated

**Review by aneeshkp — APPROVED** [review]
/lgtm

**Pull request merged**